rfcs/0001-restructure.md

```diff
@@ -0,0 +1,271 @@
+  a suitable
+  [binding](https://paketo.io/docs/buildpacks/configuration/#bindings).
+  * provides: none
+  * requires: php at `launch`
+
+* **php-redis-session-handler**:
+  Configures the given redis service instance as a PHP session
+  handler[<sup>3</sup>](#note-3). Redis settings are to be provided through a
+  suitable
+  [binding](https://paketo.io/docs/buildpacks/configuration/#bindings).
+  * provides: none
+  * requires: php at `launch`
+
+This would result in the following order groupings in the PHP language family meta-buildpack:
+
+```toml
+[[order]] # HTTPD web server
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/httpd"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-httpd"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+
+
+[[order]] # Nginx web server
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/nginx"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-nginx"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+
+[[order]] # Built-in web server (web) & FPM (php-fpm)
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-builtin-server"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+```
+
+## Environment Variables
+
+* `php-dist` buildpack must expose the location of the `php.ini` file through
```

Comment:
I feel like we can drop this env variable. If we properly install PHP, the cli should be on the path. You'll also get `php-config` which is an exec you can use to locate things like the INI file, extensions dir, etc.. so I don't think there's a reason for the buildpacks to create a non-standard way of finding similar info.

Reply:
So the `ini` file will be dropped on to the php layer by the dependency. How would the `php-config` binary or the php-installation know where this file is? Wouldn't it be in a custom named directory in the layer directory?

Reply:
Basically, the buildpacks will assume that PHP is installed and available on the path (which php-dist-cnb does). Part of what's installed is `php-config`, which you can use to query information like this. You can shell out and run `php-config` and ask about the locations of these files.

The only env variables we should set are the standard ones that tell PHP where to find the INI files.

Reply:
We basically just need to delete this line item. The php-dist cnb doesn't need to do this because other cnb's can use `php-config` to locate the php.ini file.

Reply:
@dmikusa-pivotal In the current design, downstream buildpacks do not require `php` at `build`. For them to use `php-config`, php has to be required at `build`. I can make them require `php` at build as well so they can call `php-config` and other binaries that come with php. Does that sound good?

Reply:
Hmm. I'm tempted to say yes, let's make it available at build so other cnb's can use php-config. Do you see any downside to that? Is there a performance penalty for making it required at build by more cnbs?

Reply:
I don't believe there's a penalty. Updated